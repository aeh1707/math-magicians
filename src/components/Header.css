h1 {
  width: 25vw;
  margin-right: 45vw;
  font-size: 2rem;
}

header {
  width: 85vw;
  margin: 0 auto;
  display: flex;
  align-items: center;
}

nav {
  width: 25vw;
  display: flex;
  justify-content: space-between;
}
