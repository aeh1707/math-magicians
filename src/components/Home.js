import './Home.css';
import React from 'react';
import Header from './Header';

class Home extends React.Component {
  constructor(props) {
    super(props);
    this.state = null;
  }

  render() {
    return (
      <>
        <Header />
        <main className="home">
          <h2>Welcome to our page!</h2>
          <p>
            Anim esse duis proident dolor. Eiusmod enim cupidatat nostrud quis et minim
            exercitation in incididunt amet esse aute. Sunt nostrud reprehenderit est
            reprehenderit commodo. Aute cillum aliquip enim occaecat amet. Excepteur
            minim mollit laborum fugiat. Esse sint nisi in ex consequat laborum culpa.
            Nostrud duis do dolore esse ullamco non velit ullamco deserunt sint.
          </p>
          <p>
            Anim esse duis proident dolor. Eiusmod enim cupidatat nostrud quis et minim
            exercitation in incididunt amet esse aute. Sunt nostrud reprehenderit est
            reprehenderit commodo. Aute cillum aliquip enim occaecat amet. Excepteur
            minim mollit laborum fugiat. Esse sint nisi in ex consequat laborum culpa.
            Nostrud duis do dolore esse ullamco non velit ullamco deserunt sint.
          </p>
        </main>
      </>
    );
  }
}

export default Home;
