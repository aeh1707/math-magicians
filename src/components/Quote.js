import './Quote.css';
import React from 'react';
import Header from './Header';

class Quote extends React.Component {
  constructor(props) {
    super(props);
    this.state = null;
  }

  render() {
    return (
      <>
        <Header />
        <main className="quote">
          <blockquote>
            <p>
              Carl Friedrich Gauss: &quot;Mathematics is the queen of science,
              and arithmetic the queen of mathematicss.&quot;
            </p>
          </blockquote>
        </main>
      </>
    );
  }
}

export default Quote;
