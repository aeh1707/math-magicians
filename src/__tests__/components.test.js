import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import { BrowserRouter as Router } from 'react-router-dom';
import Header from '../components/Header';
import Home from '../components/Home';
import Quote from '../components/Quote';

describe('Header renders correctly', () => {
  test('Renders Header Correctly', () => {
    const header = render(<Router><Header /></Router>);
    expect(header).toMatchSnapshot();
  });
});

describe('Home renders correctly', () => {
  test('Renders Header Correctly', () => {
    const home = render(<Router><Home /></Router>);
    expect(home).toMatchSnapshot();
  });
});

describe('Quote renders correctly', () => {
  test('Renders Header Correctly', () => {
    const quote = render(<Router><Quote /></Router>);
    expect(quote).toMatchSnapshot();
  });
});
