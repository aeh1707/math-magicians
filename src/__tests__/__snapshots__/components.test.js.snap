// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly Renders Header Correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header>
        <h1>
          Math Magicians
        </h1>
        <nav>
          <a
            href="/"
          >
            Home
          </a>
          |
          <a
            href="/Calculator"
          >
            Calculator
          </a>
          |
          <a
            href="/Quote"
          >
            Quote
          </a>
        </nav>
      </header>
    </div>
  </body>,
  "container": <div>
    <header>
      <h1>
        Math Magicians
      </h1>
      <nav>
        <a
          href="/"
        >
          Home
        </a>
        |
        <a
          href="/Calculator"
        >
          Calculator
        </a>
        |
        <a
          href="/Quote"
        >
          Quote
        </a>
      </nav>
    </header>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Home renders correctly Renders Header Correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header>
        <h1>
          Math Magicians
        </h1>
        <nav>
          <a
            href="/"
          >
            Home
          </a>
          |
          <a
            href="/Calculator"
          >
            Calculator
          </a>
          |
          <a
            href="/Quote"
          >
            Quote
          </a>
        </nav>
      </header>
      <main
        class="home"
      >
        <h2>
          Welcome to our page!
        </h2>
        <p>
          Anim esse duis proident dolor. Eiusmod enim cupidatat nostrud quis et minim exercitation in incididunt amet esse aute. Sunt nostrud reprehenderit est reprehenderit commodo. Aute cillum aliquip enim occaecat amet. Excepteur minim mollit laborum fugiat. Esse sint nisi in ex consequat laborum culpa. Nostrud duis do dolore esse ullamco non velit ullamco deserunt sint.
        </p>
        <p>
          Anim esse duis proident dolor. Eiusmod enim cupidatat nostrud quis et minim exercitation in incididunt amet esse aute. Sunt nostrud reprehenderit est reprehenderit commodo. Aute cillum aliquip enim occaecat amet. Excepteur minim mollit laborum fugiat. Esse sint nisi in ex consequat laborum culpa. Nostrud duis do dolore esse ullamco non velit ullamco deserunt sint.
        </p>
      </main>
    </div>
  </body>,
  "container": <div>
    <header>
      <h1>
        Math Magicians
      </h1>
      <nav>
        <a
          href="/"
        >
          Home
        </a>
        |
        <a
          href="/Calculator"
        >
          Calculator
        </a>
        |
        <a
          href="/Quote"
        >
          Quote
        </a>
      </nav>
    </header>
    <main
      class="home"
    >
      <h2>
        Welcome to our page!
      </h2>
      <p>
        Anim esse duis proident dolor. Eiusmod enim cupidatat nostrud quis et minim exercitation in incididunt amet esse aute. Sunt nostrud reprehenderit est reprehenderit commodo. Aute cillum aliquip enim occaecat amet. Excepteur minim mollit laborum fugiat. Esse sint nisi in ex consequat laborum culpa. Nostrud duis do dolore esse ullamco non velit ullamco deserunt sint.
      </p>
      <p>
        Anim esse duis proident dolor. Eiusmod enim cupidatat nostrud quis et minim exercitation in incididunt amet esse aute. Sunt nostrud reprehenderit est reprehenderit commodo. Aute cillum aliquip enim occaecat amet. Excepteur minim mollit laborum fugiat. Esse sint nisi in ex consequat laborum culpa. Nostrud duis do dolore esse ullamco non velit ullamco deserunt sint.
      </p>
    </main>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Quote renders correctly Renders Header Correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header>
        <h1>
          Math Magicians
        </h1>
        <nav>
          <a
            href="/"
          >
            Home
          </a>
          |
          <a
            href="/Calculator"
          >
            Calculator
          </a>
          |
          <a
            href="/Quote"
          >
            Quote
          </a>
        </nav>
      </header>
      <main
        class="quote"
      >
        <blockquote>
          <p>
            Carl Friedrich Gauss: "Mathematics is the queen of science, and arithmetic the queen of mathematics."
          </p>
        </blockquote>
      </main>
    </div>
  </body>,
  "container": <div>
    <header>
      <h1>
        Math Magicians
      </h1>
      <nav>
        <a
          href="/"
        >
          Home
        </a>
        |
        <a
          href="/Calculator"
        >
          Calculator
        </a>
        |
        <a
          href="/Quote"
        >
          Quote
        </a>
      </nav>
    </header>
    <main
      class="quote"
    >
      <blockquote>
        <p>
          Carl Friedrich Gauss: "Mathematics is the queen of science, and arithmetic the queen of mathematics."
        </p>
      </blockquote>
    </main>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
